demoapp/www/js/push.js

```diff
@@ -1,6 +1,40 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+document.addEventListener("deviceready", onDeviceReady, false);
+
+function onNotificationReceived(pushNotification) {
+    var message = pushNotification.message;
+    var title = pushNotification.title;
+
+    if (message === null || message === undefined) {
+        title = 'Android background';
+        message = '<empty>';
+    }
+
+    if (pushNotification.customProperties && Object.keys(pushNotification.customProperties).length > 0) {
+        message += '\nCustom properties:\n' + JSON.stringify(pushNotification.customProperties);
+    }
+
+    alert("title = " + title + "; message = " + message)
+}
+
+function updateNotificationListener(isEnabled) {
+
+    // This is how you can add listener for notifications.
+    if (isEnabled) {
+        AppCenter.Push.addEventListener('notificationReceived', onNotificationReceived); 
+    } else {
+        AppCenter.Push.removeEventListener('notificationReceived', onNotificationReceived); 
+    }
+}
+
+function onDeviceReady() {
+    AppCenter.Push.isEnabled(function (isEnabled) {
```

Comment:
Why are we making this check twice? Let's move the `pushEnabled` variable initialization to the place where it's first called.